skyverge/facebook-for-woocommerce
# Add product handler sync methods

# Summary

Adds some static public methods in the product handler to help setting sync flag and check the sync condition of a product.

### Story [ch26440](https://app.clubhouse.io/facebookforwoocommerce/story/26440/add-products-handler-sync-methods)

## UI changes

none

## QA

With WP CLI or WP PHP Console you can try using the following methods:

* `SkyVerge\WooCommerce\Facebook\Products::enable_sync_for_products()`: pass an array with any number of `\WC_Product[]` and you should set a post meta for each of these called `_wc_facebook_sync` with value `yes`

* `SkyVerge\WooCommerce\Facebook\Products::disable_sync_for_products()`: pass an array with any number of `\WC_Product[]` and you should set a post meta for each of these called `_wc_facebook_sync` with value `no`

* `SkyVerge\WooCommerce\Facebook\Products::is_sync_enabled_for_product()`: pass a single `\WC_Product` and you should get a `bool` returned value according to whether the meta set previously is `yes` (`true`) or `no` (`false`). If a product does not have the meta, it will be also `false`.
